client/http_post.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Spencer Kimball (spencer@cockroachlabs.com)
+//         Vivek Menezes (vivek@cockroachlabs.com)
+
+package client
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	snappy "github.com/cockroachdb/c-snappy"
+	"github.com/cockroachdb/cockroach/base"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/retry"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// PostContext is the context passed into the Post function
+type PostContext struct {
+	Server    string // The host:port address of the Cockroach gateway node
+	Endpoint  string
+	Client    *http.Client  // The HTTP client
```

Comment:
I've eliminated the client, but then I hit another problem. Without the client being created once during the initialization of the sender the code crashes. I've pushed this PR because it works, but it doesn't work without the initialization done all together. See my comment on the PR